pkg/containerdutil/helpers.go

```diff
@@ -0,0 +1,45 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package containerdutil
+
+import (
+	"context"
+
+	"github.com/containerd/containerd/content"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+var ReadBlob = readBlobWithCache()
```

Comment:
On ReadBlob specifically, it does not matter, and is safe to use in any and all cases.

Reply:
why is that? If it's imported and used by a long running process, there is no mechanism to clean up the underlying cache.

Reply:
ReadBlob only ever reads image manifest files, from their `desc.Digest`.
They are content-addressable and never change.
Even containerd allows for that behavior by returning directly from `desc.Data` if it is set, instead of actually reading the file.

In that context, the behavior of our caching mechanism is the same as if we would keep descriptors around (with the `Data` property set) - if the underlying object gets removed, your descriptor is dangling and will gladly return data for something that doesn't exist anymore.

Nevertheless, yeah, I ll add the note to document that behavior - you are right.

Thank you @djdongjin .